zeroconf/__init__.py

```diff
@@ -3168,8 +3280,7 @@ def add_listener(
                         records.append(record)
 
         if records:
-            listener.update_records(self, now, records)
-            listener.update_records_complete()
+            self.record_manager.update_engine.updates(now, records)
```

Comment:
Probably needs a new method on `record_manager` to run updates in a task and then `notify_all`

else call the notify right away